assets/js/theme/cart.js

```diff
@@ -18,18 +18,34 @@ export default class Cart extends PageManager {
         this.bindEvents();
     }
 
-    cartUpdate($target) {
+    cartUpdate($target, preVal = null) {
         const itemId = $target.data('cartItemid');
         const $el = $(`#qty-${itemId}`);
-        const oldQty = parseInt($el.val(), 10);
+        const oldQty = preVal !== null ? preVal : parseInt($el.val(), 10);
         const maxQty = parseInt($el.data('quantityMax'), 10);
         const minQty = parseInt($el.data('quantityMin'), 10);
         const minError = $el.data('quantityMinError');
         const maxError = $el.data('quantityMaxError');
-        const newQty = $target.data('action') === 'inc' ? oldQty + 1 : oldQty - 1;
+        let qty = $el.attr('value');
+        
+        switch ($target.data('action')) {
+        case 'inc':
+            qty = oldQty + 1;
+            break;
+        case 'dec':
+            qty = oldQty - 1;
+            break;
+        }
+
+        const newQty = qty;
 
         // Does not quality for min/max quantity
-        if (newQty < minQty) {
+        if (isNaN(newQty)) {
+            return swal({
+                text: "Not a valid entry",
+                type: 'error',
+            });
+        } else if(newQty < minQty) {
```

Comment:
@junedkazi I changed that to display error if there is an invalid input, however that will change the current logic that fallback in case of invalid input to the min min_purchase_quantity.  If we choose to keep it, I would like to write a test for this case

Reply:
I like this logic because I (and users) can now enter trash into the quantity field. Maybe the text should be "Not a valid number?" Because other errors will show up in the case of out of stock.